commands/slash/signin.go

```diff
@@ -142,6 +142,53 @@ func Signin() (*discordgo.ApplicationCommand, func(s *discordgo.Session, i *disc
 				entSigninType = signin.TypeOther
 			}
 
+			// 70-automatically-signin-user-who-creates-signin
+			recentSignin, err := data.Signin.RecentSignin(i.Member.User.ID, entSigninType, span.Context())
+			if err != nil {
+				logging.Error(s, err.Error(), i.Member.User, span, logrus.Fields{"error": err})
+				return
+			}
+
+			if recentSignin {
+				err = s.InteractionRespond(i.Interaction, &discordgo.InteractionResponse{
+					Type: discordgo.InteractionResponseChannelMessageWithSource,
+					Data: &discordgo.InteractionResponseData{
+						Content: "You have already signed in for **" + signinType + "**!",
+						Flags:   discordgo.MessageFlagsEphemeral,
+					},
+				})
+				if err != nil {
+					logging.Error(s, err.Error(), i.Member.User, span, logrus.Fields{"error": err})
+				}
+				return
+			}
+
+			_, err = data.Signin.Create(i.Member.User.ID, entSigninType, span.Context())
+			if err != nil {
+				logging.Error(s, err.Error(), i.Member.User, span, logrus.Fields{"error": err})
+				return
+			}
+
+			if config.Google.Enabled {
+				err = google.SheetsAppendSignin(i.Member.User.ID, i.Member.User.Username, signinType, span.Context())
+				if err != nil {
+					logging.Error(s, err.Error(), i.Member.User, span, logrus.Fields{"error": err})
+					return
+				}
+			}
+
+			err = s.InteractionRespond(i.Interaction, &discordgo.InteractionResponse{
+				Type: discordgo.InteractionResponseChannelMessageWithSource,
+				Data: &discordgo.InteractionResponseData{
+					Content: signinMessage(i.Member.User.ID, entSigninType, span.Context()),
+					Flags:   discordgo.MessageFlagsEphemeral,
+				},
+			})
+			if err != nil {
+				logging.Error(s, err.Error(), i.Member.User, span, logrus.Fields{"error": err})
+				return
+			}
```

Comment:
Message is sent to command user here

Reply:
and here: https://github.com/ritsec/ops-bot-iii/blob/70-automatically-signin-user-who-creates-signin/commands/slash/signin.go#L273-L282

Reply:
These both use the same interaction; best way to do this is just combine this into the later one